web/packages/teleterm/src/ui/DocumentTerminal/useDocumentTerminal.test.tsx

```diff
@@ -239,376 +254,310 @@ test('useDocumentTerminal shows a warning notification if the call to TerminalsS
+      expectedDocumentUpdate: {
+        serverId: serverUUID,
+        serverUri: leafServer.uri,
+        login: 'user',
+        loginHost: undefined,
+        title: 'user@foo',
+      },
+      expectedArgsOfGetServerByHostname: [leafClusterUri, 'foo'],
+    },
+    {
+      name: 'maintains incorrect loginHost with too many parts',
+      prepareDoc: doc => {
+        doc.loginHost = 'user@foo@baz';
+      },
+      mockGetServerByHostname: undefined,
+      expectedPtyCommand: {
+        kind: 'pty.tsh-login',
+        proxyHost: 'localhost:3080',
+        clusterName: 'Test',
+        login: 'user@foo',
+        serverId: 'baz',
+        rootClusterId: 'test',
+        leafClusterId: undefined,
+      },
+      expectedDocumentUpdate: {
+        serverId: 'baz',
+        serverUri: `${rootClusterUri}/servers/baz`,
+        login: 'user@foo',
+        loginHost: undefined,
+        title: 'user@foo@baz',
+      },
+      expectedArgsOfGetServerByHostname: [rootClusterUri, 'baz'],
+    },
+    {
+      // This is in order to call `tsh ssh user@foo` anyway and make tsh show an appropriate error.
+      name: 'uses hostname as serverId if no matching server was found',
+      mockGetServerByHostname: undefined,
+      expectedPtyCommand: {
+        kind: 'pty.tsh-login',
+        proxyHost: 'localhost:3080',
+        clusterName: 'Test',
+        login: 'user',
+        serverId: 'foo',
+        rootClusterId: 'test',
+        leafClusterId: undefined,
+      },
+      expectedDocumentUpdate: {
+        serverId: 'foo',
+        serverUri: `${rootClusterUri}/servers/foo`,
+        login: 'user',
+        loginHost: undefined,
+        title: 'user@foo',
+      },
+      expectedArgsOfGetServerByHostname: [rootClusterUri, 'foo'],
+    },
+    {
+      // This is the case when the user tries to execute `tsh ssh host`. We want to call `tsh ssh
+      // host` anyway and make tsh show an appropriate error. But…
+      name: 'attempts to connect even if only the host was supplied and the server was not resolved',
+      prepareDoc: doc => {
+        doc.loginHost = 'host';
+      },
+      mockGetServerByHostname: undefined,
+      expectedPtyCommand: {
+        kind: 'pty.tsh-login',
+        proxyHost: 'localhost:3080',
+        clusterName: 'Test',
+        login: undefined,
+        serverId: 'host',
+        rootClusterId: 'test',
+        leafClusterId: undefined,
+      },
+      expectedDocumentUpdate: {
+        serverId: 'host',
+        serverUri: `${rootClusterUri}/servers/host`,
+        login: undefined,
+        loginHost: undefined,
+        title: 'host',
+      },
+      expectedArgsOfGetServerByHostname: [rootClusterUri, 'host'],
+    },
+    {
+      // …it might also be the case that the username of a Teleport user is equal to a user on the
+      // host, in which case explicitly providing the username is not necessary.
+      name: 'attempts to connect even if only the host was supplied and the server was resolved',
```

Comment:
Extra test which wasn't present before. With table tests it was matter of copying the previous test case and changing `mockGetServerByHostname`.